mrp_production_allow_recursive/tests/test_mrp_production_allow_recursive.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import Command
+from odoo.tests.common import TransactionCase
+
+
+class TestMrpProductionAllowRecursive(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.category = cls.env["product.category"].create({"name": "Test Category"})
+        cls.product_finished = cls.env["product.product"].create(
+            {"name": "Finished Product"}
+        )
+        cls.product_component = cls.env["product.product"].create(
+            {"name": "Component Product"}
+        )
+        cls.bom = cls.env["mrp.bom"].create(
+            {
+                "product_tmpl_id": cls.product_finished.product_tmpl_id.id,
+                "product_qty": 1.0,
+                "bom_line_ids": [
+                    Command.create(
+                        {
+                            "product_id": cls.product_component.id,
+                            "product_qty": 1.0,
+                        }
+                    )
+                ],
+            }
+        )
+
+    def test_mrp_production_allow_recursive(self):
+        mo = self.env["mrp.production"].create(
+            {
+                "product_id": self.product_finished.id,
+                "product_qty": 1.0,
+                "bom_id": self.bom.id,
```

Comment:
Let's assume a scenario where an extra component is added to a produced item.

```suggestion
                "bom_id": False
```